src/big/big_float.cr

```diff
@@ -17,6 +18,23 @@ struct BigFloat < Float
     LibGMP.mpf_init_set_str(out @mpf, str, 10)
   end
 
+  def initialize(num : BigInt)
```

Comment:
It's part of `Object.hash` condition (that's wrong on current Crystal 0.23), read the [doc](https://crystal-lang.org/api/0.23.0/Object.html#hash-instance-method):

> This method must have the property that a == b implies a.hash == b.hash.

It's part of "change computation of hash value". Splitting of normalization and computation looks bad for me.